### DISCUSSION TITLE
What is the best practice to handling submodules that require setting the working directory in a targets pipeline?

### DISCUSSION THREAD

**Discussion by 33Vito** [ropensci/targets#1284] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hi, 

I am integrating a lot of ETL workflows under one targets repository, where each workflow has their own folder with the _targets.R file, etc. One issue I am having is that some code I use (set up as a git submodle from another repository maintained by someone else) need to set the working directory to their particular folder and uses relative folder path in their code, while the targets workflow needs to be triggered from a parent folder which has some other code in additional to the code in this submodule. I tried using `setwd()` in the `_targets.R` script without any luck. Can I borrow your wisdom here to handle such situation gracefully?

**Comment by joelnitta** [comment, marked as answer]
Perhaps `withr::with_dir()` could be of use?

https://withr.r-lib.org/reference/with_dir.html

**Reply by 33Vito** [reply]
Perfect solution, thank you so much!
